### ISSUE TITLE
ajax.aspnetcdn.com down / fallback don't work?

### ISSUE THREAD

**Issue by FredyWenger** [aspnet/Mvc#3741]
I work with VS2015 and MVC 6 RC (based on the MS standard-template) and noted problems with some links (e.g. bootstrap -glyphicon where not showed anymore in staging/production, where I had not problems in the dev-environment).
After searching for hours, I have found, that the site ajax.aspnetcdn.com seems to be down (at least for Switzerland).
Further, <strong>also the fallback seems not to work</strong> (as it should have been triggere, when the default-link cannot be accessed).
So I have deactivated the main-link, and activated the local link as standard instead and anything works again:

```html
 <environment names="Staging,Production">
        @* Deactivated, as https://ajax.aspnetcdn.com is not reachable *@
        @*<link rel="stylesheet" href="https://ajax.aspnetcdn.com/ajax/bootstrap/3.3.5/css/bootstrap.min.css"*@
        @* Instead set local reference as default *@
              <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css"
              asp-fallback-href="~/lib/bootstrap/dist/css/bootstrap.min.css"
              asp-fallback-test-class="sr-only" asp-fallback-test-property="position" asp-fallback-test-value="absolute" />
```

So...:

1. Can anybody confirm, that ajax.aspnetcdn.com down is down?
 * If yes, why can't I found any official information about on the internet (this would various sites!)
2. Why does the fallback don't work?

Thanks

**Comment by nguymin4** [comment]
@FredyWenger 
1. I met similar issue yesterday, the site is not down but it blocks CORS request to the font for glyphicon which is pretty strange. My suggest is you use different CDN ex: [cdnjs.cloudflare.com](cdnjs.cloudflare.com)
2. As I tested, the fallback-test still works

**Comment by FredyWenger** [comment]
Hey @nguymin4
Thanks for your feedback, so I know, there <strong>is</strong> a problem..
But in my case the fallback also has not worked... 
As you can see below, I have (and had) the local link already defined as fallback, but it only works, if I set the local link as main-href:
@*<link rel="stylesheet" href="https://ajax.aspnetcdn.com/ajax/bootstrap/3.3.5/css/bootstrap.min.css"*@
              <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css"
              asp-fallback-href="~/lib/bootstrap/dist/css/bootstrap.min.css"
              asp-fallback-test-class="sr-only" asp-fallback-test-property="position" asp-fallback-test-value="absolute" />
So... in my understanding, the fallback should have took place what was not the case... 
I also have seen some older notes here, that there are some bug's / typo's in the fallback code (I have started with RC and my app is based on the standard-template).
<strong>Are there some bug's in the fallback (see my code above - should the code work for you)?</strong>

Thanks

**Comment by nguymin4** [comment]
@FredyWenger 
I tested the same code you you used above and it works.
For example: If I modify the version of bootstrap to 3.3.50 which is of course not valid then there will be a `<link>` to your owned hosted bootstrap as you can see below.
![screenshot from 2015-12-11 20 28 38](https://cloud.githubusercontent.com/assets/6953273/11752122/54cd5326-a046-11e5-8d8b-3a2cf8780c29.png)

**Comment by mikes-gh** [comment]
Confirmed here today. aspnetcdn not serving glyph icons correctly.  I had to change default layout  template to use bootstrapcdn to get glyph icons working.

As an aside aspnetcdn needs to make 3.3.6 available and in bower GUI manager.  Its been out a couple of weeks.;

**Comment by pranavkm** [comment]
cc @rustd

**Comment by rustd** [comment]
We will follow up to update the CDN to have the latest version of bootstrap.

**Comment by Eilon** [comment]
Hi folks, I tried this out an the *fallback* feature seems to be working just fine. The fallback exists to support loading a local copy of Bootstrap JS, Bootstrap CSS, jQuery JS, etc. when the CDN isn't available and that's working fine. But there's no feature to do fallback for the glyphicon *fonts*.

I logged a bug in the templates repo to remove the ASP.NET CDN reference for Bootstrap because of this limitation: https://github.com/aspnet/Templates/issues/407

I've also contacted the people who run the ASP.NET CDN to make sure they're aware of this issue.

In the meantime the recommended alternatives are:
1. Use a different CDN that supports downloading the fonts. The fallback system will work with any CDN for the JS/CSS files, but not the font files.
2. Don't use a CDN for Bootstrap - just use the local Bootstrap that's in the project
